lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -2511,16 +2510,27 @@ LogicalResult ConvertAtenOp<AtenFlattenUsingIntsOp>::matchAndRewrite(
     return rewriter.notifyMatchFailure(op,
                                        "end_dim must be larger than start_dim");
 
+  SmallVector<int64_t> oldShape;
+
+  if (selfType.hasStaticShape()) {
+    for (auto s : selfType.getShape())
+      oldShape.push_back(s);
+  } else {
+    // Pushing unknown sizes
+    for (unsigned i = 0; i < selfRank; i++)
+      oldShape.push_back(ShapedType::kDynamic);
+  }
+
   SmallVector<int64_t> newShape;
-  for (auto s :
-       llvm::enumerate(makeShapeTorchCompatible(selfType.getShape()))) {
+  for (auto s : llvm::enumerate(makeShapeTorchCompatible(oldShape))) {
     int64_t idx = s.index();
     if (idx < start_dim || idx > end_dim) {
       newShape.push_back(s.value());
     } else {
       if (idx == start_dim)
         newShape.push_back(s.value());
-      else
+      // Only updating when the shapes are static
+      else if (s.value() >= 0)
```

Comment:
This does not seem right to me. If one of the dimensions in the range `[start_dim, end_dim]` is dynamic, this will result in the wrong size for the flattened portion.